### PR TITLE
Add missing LBaaS ACL and rule bindings

### PR DESCRIPTION
**Pull request by marioreggiori**
### Checklist

* [ ] added release notes to `Unreleased` section in [CHANGELOG.md](CHANGELOG.md), if user facing change

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: pkg/apis/lbaas/v1/rule_genclient.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import (
+	"context"
+	"net/url"
+
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+// EndpointURL returns the URL where to retrieve objects of type Server and the identifier of the given Server.
+// It implements the api.Object interface on *Server, making it usable with the generic API client.
+func (r *Rule) EndpointURL(ctx context.Context) (*url.URL, error) {
```
[the remark this answers is not in the document]
Comment (toothstone) [reply pkg/apis/lbaas/v1/rule_genclient.go:12]: I would be open to simply disabling this check for this function (don't know what the codeclimate syntax for that is).

Alternatively, I would consider moving the filter assembly if-chain into a helper function.

File: pkg/apis/lbaas/v1/rule_genclient.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		query := u.Query()
+		query.Add("filters", filters.Encode())
+		u.RawQuery = query.Encode()
+	}
+
+	return u, nil
+}
+
+// FilterAPIRequestBody generates the request body for Rules, replacing linked Objects with just their identifier.
+func (r *Rule) FilterAPIRequestBody(ctx context.Context) (interface{}, error) {
```
[the remark this answers is not in the document]
Comment (toothstone) [reply pkg/apis/lbaas/v1/rule_genclient.go:63]: Not sure how the line matching works here, but if the "offending" lines are the ones listing the backend and frontend attributes, I think we should just disable the check for the two/three functions.

File: pkg/apis/lbaas/v1/rule_genclient.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		filters := r.buildListFilters()
+		query := u.Query()
+		query.Add("filters", filters.Encode())
+		u.RawQuery = query.Encode()
+	}
+
+	return u, nil
+}
+
+func (r *Rule) buildListFilters() url.Values {
```
Comment (LittleFox94) [inline pkg/apis/lbaas/v1/rule_genclient.go:33]: PR #114 has a new FilterHelper which could be of great help here, maybe we want to extract that from that PR and start using it

**Review by LittleFox94 — COMMENTED** [review]
Not sure if we need some changes

File: pkg/apis/lbaas/v1/rule_types.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	ParentType       string `json:"parent_type"`
+	Index            int    `json:"index"`
+	Condition        string `json:"condition"`
+	ConditionTest    string `json:"condition_test"`
+	Type             string `json:"type"`
+	Action           string `json:"action"`
+	RedirectionType  string `json:"redirection_type"`
+	RedirectionValue string `json:"redirection_value"`
+	RedirectionCode  string `json:"redirection_code"`
+	RuleType         string `json:"rule_type"`
```
[the remark this answers is not in the document]
Comment (marioreggiori) [reply pkg/apis/lbaas/v1/rule_types.go]: I think we need to remove `omitempty` from `Index` of both `ACL` and `Rule` to allow 0 index, right?

[the remark this answers is not in the document]
Comment (LittleFox94) [reply pkg/apis/lbaas/v1/rule_types.go]: maybe we need an `OptionalInt` type, less pita to use (`Index: OptionalInt(42)`). That one could work out if it was set or is the default value

Not in lbaas/v1 package, maybe `pkg/apis/common`?

File: pkg/apis/lbaas/v1/rule_genclient.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
+package v1
```
[the remark this answers is not in the document]
Comment (LittleFox94) [reply pkg/apis/lbaas/v1/rule_genclient.go:1]: true, but let's fix that for all LBaaS resources in a fresh PR

File: pkg/apis/lbaas/v1/acl_genclient.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
+package v1
```
[the remark this answers is not in the document]
Comment (LittleFox94) [reply pkg/apis/lbaas/v1/acl_genclient.go:1]: true, but let's fix that for all LBaaS resources in a fresh PR

File: pkg/apis/lbaas/v1/rule_genclient.go (hunk 3 above)
Comment (LittleFox94) [reply pkg/apis/lbaas/v1/rule_genclient.go]: pushed a commit to use the filter helper instead, please take a look at the changes @marioreggiori 

also added tests for the filters

File: pkg/apis/lbaas/v1/rule_types.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (marioreggiori) [reply pkg/apis/lbaas/v1/rule_types.go]: I think the optional type is the way to go. But maybe like this: `pkg/apis/common/optional.Int(42)`?

File: pkg/apis/lbaas/v1/rule_genclient.go (hunk 3 above)
Comment (marioreggiori) [reply pkg/apis/lbaas/v1/rule_genclient.go]: changes LGTM

**Review by LittleFox94 — APPROVED** [review]

**Pull request merged**